### PR TITLE
updated CardType and Enumeration classes

### LINKED ISSUE #1527: Update Enumeration and CardType class
Per suggestions in [dotnet/docs](https://github.com/dotnet/docs) repository issue [#18496](https://github.com/dotnet/docs/issues/18496), update the `Microsoft.eShopOnContainers.Services.Ordering.Domain.SeedWork.Enumeration` class the and `Microsoft.eShopOnContainers.Services.Ordering.Domain.AggregatesModel.BuyerAggregate.CardType` class to use modern C# features:

- `nameof` expression
-  pattern matching
- use expression bodied members consistently

A corresponding [pull request](https://github.com/dotnet/docs/pull/21677) for the documentation is also pending.

### PR DESCRIPTION
**Pull request by mfcallahan**
PR for issue #1527 

Updating the `Microsoft.eShopOnContainers.Services.Ordering.Domain.SeedWork.Enumeration` class the and `Microsoft.eShopOnContainers.Services.Ordering.Domain.AggregatesModel.BuyerAggregate.CardType` class to use modern C# features:

- use nameof expression
- use pattern matching
- use expression bodied members consistently

A corresponding [pull request](https://github.com/dotnet/docs/pull/21677) for the Microsoft documentation is also pending; changes here are per suggestion by @sughosneo.

### CODE REVIEW

**Comment by sughosneo** [comment]
Hi @mfcallahan , thank you for submitting this PR. Will test and update.

**Comment by sughosneo** [comment]
> Hi @mfcallahan , thank you for submitting this PR. Will test and update.

Hi @mfcallahan , if you could update the PR with the changes suggested by @IEvangelist

Thanks.

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (IEvangelist) [inline src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]:
```suggestion
                     .Select(f => f.GetValue(null))
                     .Cast<T>();
```

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (IEvangelist) [inline src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]:
```suggestion
            if (obj is not Enumeration otherValue)
```

**Review by IEvangelist — COMMENTED** [review]
Looks like these were lost somehow, adding them again here.

**Comment by sughosneo** [comment]
> Looks like these were lost somehow, adding them again here.

Thank you @IEvangelist

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (mfcallahan) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: This will require bumping the project to C# 9.0 - is that OK for the scope of this issue?

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (mfcallahan) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: Updated.

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (sughosneo) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: Hi @mfcallahan, we are in process of updating this project to .NET 5 and C# 9.0. It should be available soon. 
So to my opinion, it would be better to use C# 9.0 feature itself, because it would save some rework in both from docs and reference app point of view. We will merge this PR once the `dev` branch is updated and performed the integration testing then. Thoughts ?

**Comment by IEvangelist** [comment]
Hi @mfcallahan - I noticed that you manually pushed the suggested changes, so I'm not sure if you were aware of this or not. But you could actually accept suggested changes from GitHub - and you could "batch" suggestions and commit them all at once. This is the preferred way as it adds the reviewer who made the suggestion a co-author in the commit history, and removes the unnecessary manual effort of revolving review comments.

https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

**Comment by mfcallahan** [comment]
Thanks for the feedback @IEvangelist, how would you like me to proceed with this PR? I can revert the manual changes and use the "Commit suggestion" option if that is preferred.

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (mfcallahan) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: Sounds good, I will commit @IEvangelist's suggestion here and use the newer C# feature knowing the project is in the process of being updated and this pull request will be merged at that time.

Comment (IEvangelist) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: We should really try to keep this at least cross-referenced... https://github.com/dotnet/docs/pull/21677

**Comment by IEvangelist** [comment]
> Thanks for the feedback @IEvangelist David Pine FTE, how would you like me to proceed with this PR? I can revert the manual changes and use the "Commit suggestion" option if that is preferred.

No, not a big deal. I was simply trying to let you know about this feature, to make things easier for future PRs.

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (sughosneo) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: Yes, that's right. I have updated the label also to `.NET 5.0` for better clarity.

**Review by AndriiPavliuk — APPROVED** [review]

File: src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs
Comment (sughosneo) [reply src/Services/Ordering/Ordering.Domain/SeedWork/Enumeration.cs]: Hi @mfcallahan, `dev` branch is updated to `.NET 5` now. Could you please commit the suggestion and update this PR ? Will test and merge in the changes.

Thank you.

**Review by sughosneo — APPROVED** [review]
Tested the changes after merging it locally with dev. LGTM!

**Pull request merged**
